python/ray/data/impl/arrow_block.py

```diff
@@ -80,6 +80,9 @@ def __init__(self):
 
     def add(self, item: Any) -> None:
         if self._builder is None:
+            # TODO(ekl) this isn't efficient, but makes the identity map work
+            if isinstance(item, ArrowRow):
+                item = item.as_pydict()
```

Comment:
Why not give the `ArrowRow` directly to `check.add(item)` below?

Reply:
Done